tensorflow/core/common_runtime/eager/mkl_eager_op_rewrite.cc

```diff
@@ -0,0 +1,185 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifdef INTEL_MKL
+#include "tensorflow/core/common_runtime/eager/eager_op_rewrite_registry.h"
+#include "tensorflow/core/graph/mkl_graph_util.h"
+#include "tensorflow/core/graph/mkl_layout_pass.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/util/mkl_util.h"
+#include "tensorflow/core/util/util.h"
+
+namespace tensorflow {
+
+class MklEagerOpRewrite : public EagerOpRewrite {
+ public:
+  MklEagerOpRewrite(string name, string file, string line);
+  typedef struct {
+    string op_name;
+    std::function<bool(EagerOperation*)> RewriteRule;
+    std::function<Status(EagerOperation*, std::unique_ptr<EagerOperation>*)>
+        CreateMklOp;
+  } MklEagerOp;
+
+ private:
+  std::vector<MklEagerOp> mkl_eager_ops;
```

Comment:
Should this be an unordered map instead?

Reply:
Technically yes.
However, this coding style (using "vector" is consistent with that of  core/graph/mkl_layout_pass.cc which involves most MKL op rewritting. 
Because the size of the vector is only 3; so not much performance impact. 

Thanks!

Reply:
Don't you plan on adding more ops/rewrite rules eventually? I think `unordered_map` is better, and we don't need to keep this consistent with the rewrite info (`rinfo_`) in `mkl_layout_pass.cc`. As a matter of fact, I also think that `rinfo_` should be `unordered_map`.

Also, per our style guide, class member names should end with "_" (`mkl_eager_ops_`).

Reply:
Done with following
  1. renamed mkl_eager_ops by appending "_"
  2. Add a TODO comment in source file for future refactoring per suggestion